### ISSUE TITLE
Inquiry about training configuration for stage 1 of DocParser

### ISSUE THREAD

**Issue by duongttr** [NormXU/DocParser-Pytorch#3]
Hi,

I have just read the paper "DocParser" and was disappointed to find that there is no official code provided by the authors. Fortunately, I came across your code, which saved my day.

However, I came across a section in the paper that explains how to pre-train the first stage (Knowledge Transfer step), and it has left me confused. The paper mentions that the features from the ConvNeXt-based model are mapped with the OCR model using L2 Loss. This is perplexing because the OCR model is pre-trained on text crops, while ConvNeXt requires pre-training on full document images. Additionally, the image sizes of these two models do not appear to match. Could you please help me understand how this stage is configured?

I understand that providing detailed information about the training configuration might be time-consuming, but any insights you can provide would greatly help me in understanding the methodology more clearly.

Thank you for your time and consideration. I look forward to your response.

**Comment by NormXU** [comment]
@duongttr I agree with you that this part is quite confusing. I reached out to the author and tried to implement it under his guidance:

Consider the ConvNext as an image encoder, whose input is $H$, $W$, and output is $H/32$, $W/8$. The dataset for distillation consists of text-image pairs cropped from the document, with each image containing only one line of text.

Then, we have two options here:

1. We can directly train the ConvNext + BiLSTM with CTC Loss using our text-image dataset if its size is large enough. The size of the text image chunk should be resized to $32 \times W$. This is important since ConvNext can reduce the height to $H/32$, which is exactly $1$. Therefore, we can regard the output features as time sequences, whose size is $1 \times W/8$
 
Or

3. We can distill the output features from the image backbone of PP-OCR-V2. To obtain the same dimension output, we need a point-wise convolution layer initialized randomly.

I took option 1 since I already have a complete OCR training pipeline and dataset.

Hope this can answer your question.

**Comment by duongttr** [comment]
@NormXU Thank you very much for your explanation. I have just one more question regarding option 1.
I believe your option 1 is reasonable, but does the resizing of the text-image chunk to $32 \times W$ affect the subsequent step of the training stage (Masked Document Reading), considering that the required image size (as mentioned in the paper) for the overall pre-trained model is $2560 \times 1920$?

**Comment by NormXU** [comment]
@duongttr We expect the Swin-Transformer part to conduct the overall feature extraction. The ConvNext part only do the local/intra feature extractions between texts.

**Comment by duongttr** [comment]
@NormXU Thank you very much. Your explanation has been very helpful. I am now going to close the issue.

**Comment by Mohamed-Dhouib** [comment]
One of the authors of the paper here and I want to add some clarifications:

- CNNs are shift-equivariant. This means that if the input to a convolutional layer is shifted, the output feature maps will also shift by the same amount, except for the extreme regions of the image. Therefore, if we feed a full image to the recognition backbone of any Optical Character Recognition system, the output features will still be meaningful.
We verified this in our initial experiments where we used a frozen OCR recognition network , followed only by two convolutional layers, as a backbone for our model and achieved satisfactory results (we tried EasyOCR and different versions of PaddleOCR). Hence, training the ConvNext part to produce features similar to the PP-OCR recognition backbone (with a pointwise convolution to align dimensions) will enable the model to extract relevant features.

- The Knowledge Transfer step is implemented to speed up pretraining. So, even if it's not performed or replaced by a simpler option (as explained by @NormXU), it should still work.

- If your resources are limited, you can use a resolution lower than 2560 * 1920. It also depends on your pretraining dataset. The key idea here is to avoid a resolution discrepancy between your pretraining and downstream tasks which you can simply do with data augmentation (in our case we had only the text for the pretraining images (no bounding boxes), so zooming in to make text bigger was not an option as we can't eliminate the cropped text from the ground truth). You can get relevant OCR predictions for the ild dataset from this repo : https://github.com/furkanbiten/idl_data
